### DISCUSSION TITLE
MTEX ORTOOLS  variant of K-S

### DISCUSSION THREAD

**Discussion by wanghaijian217** [ORTools4MTEX/ORTools#54] [Q&A]
Hello, Frank!

When ORTOOLS is reconstructed, we rely on K-S, N-W or other orientation relations, and on this basis, job.calcParent2Child;; Make corrections. Then, the problem arises. The standard K-S relationship defines 24 variants. After I shear them according to my own method, the orientation relationship of variants is no longer V1 ~ V24 defined by Morito et al. How can I modify them? Introduce the orientation relationship of each variant in my K-S relationship. My K-S variant orientation relationship is as follows: from left to right, the table is variant serial number, shear plane, first shear direction, second shear direction, strain matrix, direction invariant crystal plane and eigenvalue, habitus plane.
best wishes!!
![image](https://github.com/ORTools4MTEX/ORTools/assets/131143952/ac0fc63c-5f52-465c-9cfe-f81006bcdd49)
![image](https://github.com/ORTools4MTEX/ORTools/assets/131143952/ca9cca3c-3e51-4d49-9d15-79b8d4cea947)

**Comment by AzdiarGazder** [comment]
Hi @wanghaijian217,

Before this new issue is taken up, could you please confirm if the previous issue from yesterday was resolved?
If so, could you please mark the reply as the answer so that we can close the discussion?
If not, please explain how we could better resolve your problem.

Also, when requesting debugging, please:
- provide all details, and
- upload your script (and if needed, your ebsd map data) as well.

Warm regards,
Azdi

**Reply by wanghaijian217** [reply]
Hi Azdiar
Sorry, I didn't notice the mark as answer button before. Yesterday's problem has been solved. As you can see, the status has been updated.
Now my question is how to import my above calculation data into K-S relationship. Because the commonly used variant analysis in K-S relationship is divided according to the following document or Morito et al. 's document: 10.1016/j.actamat.2011. 12.018. I want to know how to import my partition of variants into your ORTOOLS for refactoring?

**Reply by AzdiarGazder** [reply]
Hi @wanghaijian217,

Thank you for marking my reply as the answer in our previous discussion.
It enables us as maintainers to close discussions with users as *resolved*.

If you have written a script that calculates these values, please upload it here.
That way, I can work out how to integrate your calculations into MTEX and ORTools in a more generic manner.
If needed, please upload your ebsd map data as well.

Warm regards,
Azdi
